source/packages/modulo_core/include/modulo_core/Communication/MessagePassing/ReadStateConversion.hpp

```diff
@@ -148,13 +151,25 @@ void read_msg(state_representation::DualQuaternionTwist& state, const geometry_m
 
 /**
  * @brief Template function to convert a ROS std_msgs::msg::T to a Parameter<T>
- * @param state The Parameter<T> to populate
- * @param msg The ROS msg to read from
  * @tparam T all types of parameters supported in ROS std messages
  * @tparam U all types of parameters supported in ROS std messages
+ * @param state The Parameter<T> to populate
+ * @param msg The ROS msg to read from
  */
 template <typename T, typename U>
 void read_msg(state_representation::Parameter<T>& state, const U& msg) {
   state.set_value(msg.data);
 }
+
+/**
+ * @brief Convert a ROS std_msgs::msg::ByteMultiArray to a State using protobuf decoding
+ * @tparam a state_representation::State type object
+ * @param state The state to populate
+ * @param msg The ROS msg to read from
+ */
+template <typename T>
+void read_msg(T& state, const std_msgs::msg::ByteMultiArray& msg) {
```

Comment:
For consistency maybe use `modulo::core::EncodedState` here, and in WriteState also... then you can include `Communication/EncodedState.hpp` instead of `std_msgs/msg/byte_multi_array.hpp`, and it perhaps stays a little easier to maintain